clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/converter/manifest/KubernetesDeployManifestConverter.java

```diff
@@ -50,13 +62,61 @@ public AtomicOperation convertOperation(Map input) {
 
   @Override
   public KubernetesDeployManifestDescription convertDescription(Map input) {
-    return (KubernetesDeployManifestDescription)
-        KubernetesAtomicOperationConverterHelper.convertDescription(
-            input, this, KubernetesDeployManifestDescription.class);
+    KubernetesDeployManifestDescription mainDescription =
+        (KubernetesDeployManifestDescription)
+            KubernetesAtomicOperationConverterHelper.convertDescription(
+                input, this, KubernetesDeployManifestDescription.class);
+    return convertListDescription(mainDescription, getObjectMapper());
   }
 
   @Override
   public boolean acceptsVersion(ProviderVersion version) {
     return version == ProviderVersion.v2;
   }
+
+  /**
+   * If present, converts a KubernetesManifest of kind List into a list of KubernetesManifest
+   * objects.
+   *
+   * @param mainDescription deploy manifest description as received.
+   * @param objectMapper used to generate the input deploy manifest description.
+   * @return updated description.
+   */
+  @SuppressWarnings("unchecked")
+  private KubernetesDeployManifestDescription convertListDescription(
```

Comment:
I'm wondering whether this would live better in the actual `KubernetesDeployManifestOperation` class.  In general the converter is intended to just deserialize the operation with more complex logic being handled in the actual operation itself.  Anything else that requires actually looking at the manifests and manipulating them (such as looking up namespaces, etc) is in the operation.

Reply:
That's also another possibility, `operate` function already performs some initial logic over the list of input `KubernetesManifest` objects. I don't know however if there are other side effects of delaying manifest splitting until the operation actually executes, but I can test if there are no problems related with caching for instance. 

A drawback is that there's no `ObjectMapper` in the operation, then we would end up having two different paths for converting an input `Map` from the request into `KubernetesManifest`: one in the converter used for everything and other in the operation used only when resources are grouped in `List` manifests. If the way that deserializing manifests changes (i.e. using a custom ObjectMapper), there would be two places to change and the `List` one would not be obvious. That said, I have no problem moving the split to the operation itself.

Reply:
@ezimanyi given the above, do you think it's better to move this logic to `KubernetesDeployManifestOperation`?

Reply:
Yeah, as I think of it more, it probably makes sense to have all of the unwrapping of manifests happen in the same place---both so that they can use the same `objectMapper` but also so that downstream logic can just assume it has a list of manifests to deploy.  I think it's fine to leave in this file.